src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql

```diff
@@ -1366,7 +1388,7 @@ AS
     END
 
 
-    IF (@etag IS NOT NULL AND @etag <> (@version - 1)) BEGIN
+    IF (@etag IS NOT NULL AND @etag <> @previousVersion) BEGIN
```

Comment:
Should we do this check earlier, before the updates/deletes?

Reply:
Yeah, we should.